src/map.c

```diff
@@ -64,14 +64,7 @@ SEXP map_impl(SEXP env, SEXP type_, SEXP progress, SEXP error_call) {
   SEXPTYPE type = Rf_str2type(CHAR(Rf_asChar(type_)));
 
   SEXP x_val = PROTECT(Rf_eval(x, env));
-
   int n = Rf_length(x_val);
-  if (n == 0) {
-    SEXP out = PROTECT(Rf_allocVector(type, 0));
-    copy_names(x_val, out);
-    UNPROTECT(2);
-    return out;
-  }
 
   // Constructs a call like f(x[[i]], ...) - don't want to substitute
   // actual values for f or x, because they may be long, which creates
```

Comment:
Unrelated comment

I had a thought that in line 112 of `pmap()`, rather than do this dance:

```c
int n = m == 0 ? 0 : Rf_length(VECTOR_ELT(l_val, 0));
```

you could just call this in `pmap_()` to get the common size, then pass it on to C

```r
size <- vec_size_common(!!!.l, .arg = ".l", .call = ..error_call)
.l <- vec_recycle_common(!!!.l, .arg = ".l", .call = ..error_call, .size = size)
```

Reply:
Could also do a similar thing in `map2_()`

Reply:
Oh hey if you also passed the size through in `map_()` then you'd avoid needing length-dispatch altogether!

(Probably still need names-dispatch)

Reply:
Oh but we could also just set the names in `map_` etc.

Reply:
Oh that's so much simpler 😆